src/um/dsound.rs

```diff
@@ -185,3 +185,57 @@ pub const DSBPLAY_LOCSOFTWARE: DWORD = 0x00000004;
 pub const DSBPLAY_TERMINATEBY_TIME: DWORD = 0x00000008;
 pub const DSBPLAY_TERMINATEBY_DISTANCE: DWORD = 0x000000010;
 pub const DSBPLAY_TERMINATEBY_PRIORITY: DWORD = 0x000000020;
+extern "system" {
+    pub fn DirectSoundCreate(
+        pcGuidDevice: LPCGUID,
+        ppDS: *mut LPDIRECTSOUND,
+        pUnkOuter: LPUNKNOWN,
+    ) -> HRESULT;
+    // pub fn DirectSoundEnumerateA(
+    //     pDSEnumCallback: LPDSENUMCALLBACKA,
+    //     pContext: LPVOID,
+    // ) -> HRESULT;
+    // pub fn DirectSoundEnumerateW(
+    //     pDSEnumCallback: LPDSENUMCALLBACKW,
+    //     pContext: LPVOID,
+    // ) -> HRESULT;
+    // pub fn DirectSoundCaptureCreate(
+    //     pcGuidDevice: LPCGUID,
+    //     ppDSC: *mut LPDIRECTSOUNDCAPTURE,
+    //     pUnkOuter: LPUNKNOWN,
+    // ) -> HRESULT;
+    // pub fn DirectSoundCaptureEnumerateA(
+    //     pDSEnumCallback: LPDSENUMCALLBACKA,
+    //     pContext: LPVOID,
+    // ) -> HRESULT;
+    // pub fn DirectSoundCaptureEnumerateW(
+    //     pDSEnumCallback: LPDSENUMCALLBACKW,
+    //     pContext: LPVOID,
+    // ) -> HRESULT;
+    // pub fn DirectSoundCreate8(
+    //     pcGuidDevice: LPCGUID,
+    //     ppDS8: *mut LPDIRECTSOUND8,
+    //     pUnkOuter: LPUNKNOWN,
+    // ) -> HRESULT;
+    // pub fn DirectSoundCaptureCreate8(
+    //     pcGuidDevice: LPCGUID,
+    //     ppDSC8: *mut LPDIRECTSOUNDCAPTURE8,
+    //     pUnkOuter: LPUNKNOWN,
+    // ) -> HRESULT;
+    // pub fn DirectSoundFullDuplexCreate(
+    //     pcGuidCaptureDevice: LPCGUID,
+    //     pcGuidRenderDevice: LPCGUID,
+    //     pcDSCBufferDesc: LPCDSCBUFFERDESC,
+    //     pcDSBufferDesc: LPCDSBUFFERDESC,
+    //     hWnd: HWND,
+    //     dwLevel: DWORD,
+    //     ppDSFD: *mut LPDIRECTSOUNDFULLDUPLEX,
+    //     ppDSCBuffer8: *mut LPDIRECTSOUNDCAPTUREBUFFER8,
+    //     ppDSBuffer8: *mut LPDIRECTSOUNDBUFFER8,
+    //     pUnkOuter: LPUNKNOWN,
+    // ) -> HRESULT;
```

Comment:
Yes, all of these rely on structs that haven't been defined. Since they were like this in the original file, I decided to keep them in the new ported file.